snap/hooks/install

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -eux
+
+# Create the config file that is used by pbm
+PBM_CONFIG_FILE="${SNAP_COMMON}/pbm_config.yaml"
+touch "${PBM_CONFIG_FILE}"
+
+# the pbm-agent daemon is not run as root meaning it does not have access write to SNAP_DATA, so
+# we need to create a child dir and give it access.
+mkdir -p "${SNAP_DATA}/pbm-agent-dir"
+
+# Copy over the mongod.conf and create needed directories/files
+MONGO_CONFIG_FILE="${SNAP_COMMON}/mongod.conf"
+echo "configuration file does not exist."
+echo "copying default config to ${MONGO_CONFIG_FILE}"
+cp -r ${SNAP}/etc/mongod.conf ${SNAP_COMMON}
+mkdir -p ${SNAP_COMMON}/db
+touch ${SNAP_DATA}/mongod.log
+touch ${SNAP_DATA}/mongod.pid
+
+
+sed -i "s/fork: true/fork: false/g" $SNAP_COMMON/mongod.conf
+sed -i "s:/var/log/mongodb/mongod.log:$SNAP_DATA/mongod.log:g" $SNAP_COMMON/mongod.conf
+sed -i "s:/var/lib/mongodb:$SNAP_COMMON/db:g" $SNAP_COMMON/mongod.conf
+sed -i "s:/var/run/mongod.pid:$SNAP_DATA/mongod.pid:g" $SNAP_COMMON/mongod.conf
+
+# Change ownership of snap directories to allow snap_daemon to read/write
+chown -R 584788:root ${SNAP_DATA}/*
+chown -R 584788:root ${SNAP_COMMON}/*
```

Comment:
JFMI only, why we hardcode magic number `584788` instead of `snap_daemon`?

Reply:
It's for giving flexibility in the rock by not specifying the exact username.  This allows us to use the `mongo` user in the script by setting the user id manually for both accounts.